### PR TITLE
perf: Fix comments double rendering on mount

### PR DESCRIPTION
**Pull request by tommoor**
There's more to improve here, but this is one of the main culprits.

Towards #7823

### CODE REVIEW

File: app/scenes/Document/components/CommentThreadItem.tsx [hunk 1]
```diff
@@ -117,31 +116,24 @@ function CommentThreadItem({
...
   const handleCancel = () => {
-    setData(toJS(comment.data));
+    setData(comment.data);
     setReadOnly();
-    setForceRender((i) => ++i);
   };
 
-  React.useEffect(() => {
-    setData(toJS(comment.data));
-    setForceRender((i) => ++i);
```
Comment (tommoor) [inline app/scenes/Document/components/CommentThreadItem.tsx:142]: This was causing double render

File: app/scenes/Document/components/CommentThreadItem.tsx [hunk 2]
```diff
@@ -117,31 +116,24 @@ function CommentThreadItem({
...
 
   const handleCancel = () => {
-    setData(toJS(comment.data));
+    setData(comment.data);
     setReadOnly();
-    setForceRender((i) => ++i);
   };
 
-  React.useEffect(() => {
-    setData(toJS(comment.data));
```
Comment (tommoor) [inline app/scenes/Document/components/CommentThreadItem.tsx:141]: With `observable.shallow` there is no need to use `toJS` in this component

Comment (hmacr) [reply app/scenes/Document/components/CommentThreadItem.tsx:141]: Looks to be the main cause 👌

**Comment by hmacr** [comment]
Way more responsive than before 💯

**Review by hmacr — APPROVED** [review]

**Comment by NangeGao** [comment]
When I open a page, the comment list flashes and then disappears. The buttons("Share", "Edit" and "New doc") in the upper right corner also disappears. We have a lot of data now. When I scroll the left collection navigation menu until the current article is displayed, the comment list is rendered out again.
Is this also related to this?

**Comment by tommoor** [comment]
This doesn't sound related, what version are you running? Make sure it's updated

**Comment by NangeGao** [comment]
> This doesn't sound related, what version are you running? Make sure it's updated

v0.79.1
You need to find one doc whose collection is ranked after 100th in the left navigation menu. Then open the comments sidebar, refresh the page, and scroll the left navigation menu to the location of the article. It will reproduce.

**Comment by NangeGao** [comment]
> > This doesn't sound related, what version are you running? Make sure it's updated
> 
> v0.79.1 You need to find one doc whose collection is ranked after 100th in the left navigation menu. Then open the comments sidebar, refresh the page, and scroll the left navigation menu to the location of the doc. It will reproduce.

I have updated to v0.80.2, the problem remains.

**Comment by tommoor** [comment]
Please file a new issue with the reproduction steps, thanks

**Comment by NangeGao** [comment]
> Please file a new issue with the reproduction steps, thanks

I was on vacation a few days ago. I can't reproduce this bug today. I will create an issue later if there is any problem. Thank you.

**Pull request merged**